# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tsa/slrn/slrn-0.9.8.0-ranges.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- slrn-0.9.8.0/src/ranges.c	2003-08-02 01:19:39.000000000 +0200
+++ slrn/src/ranges.c	2003-09-02 14:45:34.000000000 +0200
@@ -264,13 +264,17 @@
    do
      {
 	/* skip ranges that don't intersect at all */
-	if (b != NULL)
-	  while ((a != NULL) && (a->max < b->min))
-	    a = a->next;
-	
-	if (a != NULL)
-	  while ((b != NULL) && (b->max < a->min))
-	    b = b->next;
+	do
+	  {
+	     if (b != NULL)
+	       while ((a != NULL) && (a->max < b->min))
+		 a = a->next;
+	     
+	     if (a != NULL)
+	       while ((b != NULL) && (b->max < a->min))
+		 b = b->next;
+	  }
+	while ((a!=NULL) && (b!=NULL) && (a->max < b->min));
 	
 	/* append a range containing the next intersection */
 	if ((a!=NULL) && (b!=NULL))
